File: codeforces/271A/py/271a.py

def is_beautiful(y):
    have = [0] * 10
    while y != 0:
        mod = y % 10
        if have[mod]:
            return False
        else:
            have[mod] = True
        y = int(y / 10)

    return True

year = int(input())

i = 0
for i in range(year+1, 10000):
    if is_beautiful(i):
        break
print(i)

File: codeforces/271A/cpp/271a.cpp
#include <iostream>
using namespace std;

bool is_beautiful(int y)
{
    bool have[10] = {false};

    while (y) {
        int mod = y % 10;
        if (have[mod]) {
            return false;
        }
        else {
            have[mod] = true;
        }
        y /= 10;
    }

    return true;
}

int main()
{
    int year;
    cin >> year;

    int result = 0;
    for (int i = year + 1; i < 10000; i++) {
        if (is_beautiful(i)) {
            result = i;
            break;
        }
    }

    cout << result << endl;
    return 0;
}
